hpx/runtime/actions/basic_action.hpp

```diff
@@ -271,11 +271,10 @@ namespace hpx { namespace actions
         ///////////////////////////////////////////////////////////////////////
         typedef typename traits::is_future<local_result_type>::type is_future_pred;
 
-        template <typename LocalResult>
         struct sync_invoke
         {
             template <typename IdOrPolicy, typename ...Ts>
-            HPX_FORCEINLINE static LocalResult call(
+            HPX_FORCEINLINE static local_result_type call(
                 boost::mpl::false_, launch policy,
```

Comment:
The Problem here is that is_future returns a boost::mpl type. I wanted to
avoid changing too much unrelated stuff in one PR.